### PR TITLE
[Feature] Add support for the focal Tversky loss

### PR DESCRIPTION
**Pull request by zifuwanggg**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily get feedback. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation

The focal Tversky loss was proposed in https://arxiv.org/abs/1810.07842. It has nearly 600 citations and has been shown to be extremely useful for highly imbalanced (medical) datasets. To add support for the focal Tversky loss, only few lines of changes are needed for the Tversky loss.

## Modification

Add `gamma` as (optional) argument in the constructor of `TverskyLoss`. This parameter is then passed to `tversky_loss` to compute the focal Tversky loss. 

## BC-breaking (Optional)

Does the modification introduce changes that break the backward-compatibility of the downstream repos?
If so, please describe how it breaks the compatibility and how the downstream projects should modify their code to keep compatibility with this PR.

## Use cases (Optional)

If this PR introduces a new feature, it is better to list some use cases here, and update the documentation.

## Checklist

1. Pre-commit or other linting tools are used to fix the potential lint issues.
2. The modification is covered by complete unit tests. If not, please add more unit test to ensure the correctness.
3. If the modification has potential influence on downstream projects, this PR should be tested with downstream projects, like MMDet or MMDet3D.
4. The documentation has been modified accordingly, like docstring or example tutorials.

Reopening of previous [PR](https://github.com/open-mmlab/mmsegmentation/pull/2783).

### CODE REVIEW

File: mmseg/models/losses/tversky_loss.py [hunk 1]
```diff
@@ -87,15 +93,18 @@ def __init__(self,
...
                  beta=0.7,
+                 gamma=1.0,
                  loss_name='loss_tversky'):
         super(TverskyLoss, self).__init__()
         self.smooth = smooth
         self.class_weight = get_class_weight(class_weight)
         self.loss_weight = loss_weight
         self.ignore_index = ignore_index
         assert (alpha + beta == 1.0), 'Sum of alpha and beta but be 1.0!'
+        assert gamma >= 1.0, 'gamma should be at least 1.0!'
```
Comment (xiexinch) [inline mmseg/models/losses/tversky_loss.py:104]: Since there is an assertion statement, we might add a unit test for it.

File: mmseg/models/losses/tversky_loss.py [hunk 2]
```diff
@@ -31,6 +32,8 @@ def tversky_loss(pred,
                 alpha=alpha,
                 beta=beta,
                 smooth=smooth)
+            if gamma > 1.0:
+                tversky_loss **= gamma
```
Comment (xiexinch) [inline mmseg/models/losses/tversky_loss.py:36]: Should it be `1 / gamma`?
At the official implementation https://github.com/nabsabraham/focal-tversky-unet/blob/master/losses.py#L67, they calculate the FTL with `gamma=1/(4/3)=0.75`

**Review by xiexinch — COMMENTED** [review]
Hi @zifuwanggg,
Thanks for your contribution! We really appreciate it. Here are some comments that should be resolved, please take a view.

File: mmseg/models/losses/tversky_loss.py (hunk 2 above)
Comment (zifuwanggg) [reply mmseg/models/losses/tversky_loss.py]: Yes indeed. I've modified it to `1 / gamma`.

File: mmseg/models/losses/tversky_loss.py [hunk 3]
```diff
@@ -75,6 +78,9 @@ class TverskyLoss(nn.Module):
         beta (float, in [0, 1]):
             The coefficient of false negatives. Default: 0.7.
             Note: alpha + beta = 1.
+        gamma (float, in [1, inf]): The focal term. When `gamma` > 1,
+            the loss focuses more on less accurate predictions that
+            have been misclassified. Default: 1.0.
```
[the remark this answers is not in the document]
Comment (zifuwanggg) [reply mmseg/models/losses/tversky_loss.py:87]: I've added the reference in the docstring.

File: mmseg/models/losses/tversky_loss.py (hunk 1 above)
Comment (zifuwanggg) [reply mmseg/models/losses/tversky_loss.py:108]: I presume I don't need to do anything for this comment?
